plugins/spu2-x/src/Windows/Config.cpp

```diff
@@ -262,6 +262,17 @@ BOOL CALLBACK ConfigProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
             swprintf_s(temp, L"%d%%", configvol);
             SetWindowText(GetDlgItem(hWnd, IDC_VOLUME_LABEL), temp);
 
+            const int module = (int)SendMessage(GetDlgItem(hWnd, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
+            bool IsConfigurable = module != 0;
+            bool XAudio2Expansion = module == 1 || module == 4;
```

Comment:
Variable name and hardcoded values haven't been resolved (https://github.com/PCSX2/pcsx2/pull/1963#discussion_r122554260).

Also could use `const bool`.

Reply:
It looks like I'll have to use constant bool because It doesn't look like I can convert the values
 values any other way. 

```const bool XAudio2Expansion = module == 1 || module == 4;```
So would this be O.K. besides the variable name? (I'll change that later)

Reply:
I copy my previous response:
> Well, what does `1` and `4` refer to? They may currently refer to XAudio2 and PortAudio, but if, for instance, someone removes the DirectSound or WaveOut module or add a new Windows module, then that'll likely change.

Reply:
For handling hard coded values - I'd prefer if enumerations are use over macros.

Reply:
@turtleli I'm still stuck on how to get rid of the hardcoded values. This works, but I can't use logical OR with it and I'm pretty sure it's the same as before.

```
	const int OutputModule = (int)SendMessage(GetDlgItem(window, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
	const bool AudioExpansion = mods[OutputModule] != XAudio2Out;
	const bool IsConfigurable = mods[OutputModule] == XAudio2Out || PortaudioOut || WaveOut || DSoundOut;
```

Reply:
This is wrong on avery basic level. '==' binds stronger than '||'. So the first part of your conditoonal chain will be either true or false and afterwards you OR them with PortAudioOut, WaveOut and so on which are numbers. Only combine true or false with AND and OR...

What you really want to do is check whether the module is xaudio or the module is portaudio or the module is waveout and so on.

Reply:
Yeah, as willkuer said, you've made a very basic mistake.
```
const bool IsConfigurable = mods[OutputModule] == XAudio2Out || PortaudioOut || WaveOut || DSoundOut;
```
is essentially the same as:
```
const bool IsConfigurable = (mods[OutputModule] == XAudio2Out) || (PortaudioOut != nullptr)
    || (WaveOut != nullptr) || (DSoundOut != nullptr);
```
which unconditionally evaluates to true, and I kinda doubt that's intended.

Note that `XAudio2Out` only covers Windows 8+. `XAudio2_27_Out` covers earlier Windows versions.

Reply:
OK, so I assume this method can't be used at all right?

On a side note, I already have this working inside a function. I'm also going to see if it's worth putting some of the slider stuff into a function as well (it seems a little copy and paste) but If I can't figure out a way to change these hardcoded values; it may be best that you take over the rest of the PR ;)

Reply:
Disclaimer: What I wrote before about these DSoundOut, WaveOut, ... being numbers was wrong. I thought they are enums as I didn't read the code. As turtleli pointed out they are pointers (of type SndOutModule*) so you don't OR numbers but pointers with the first initial bool (which leads to the line turtleli posted). What you however want to do is OR bools. Each bool checks whether the current module is pointing to the DSoundOut, WaveOut, ... module.


Instead of 
```
const bool IsConfigurable =
    (mods[OutputModule] == XAudio2Out) ||
    (PortaudioOut != nullptr) ||
    (WaveOut != nullptr) ||
    (DSoundOut != nullptr);
```

you want to know

```
const bool IsConfigurable =
    (mods[OutputModule] == XAudio2Out) ||
    (mods[OutputModule] == PortaudioOut) ||
    (mods[OutputModule] == WaveOut) ||
    (mods[OutputModule] == DSoundOut);
```

Btw. this is a really nice example why OOP is a good concept for module programming. This could be handled in appropriate child classes without looking that wonky.

Reply:
@MrCK1 Some things I forgot to mention in my previous comment - When you check whether it's configurable or not, I don't get why you don't just check that it's not the null output module. It's what you initially did as well. And for audio expansion, you seem to have omitted PortAudio again. As for fixing the logic, just read what willkuer posted and adapt.